lib/floe/workflow/intrinsic_function.rb

```diff
@@ -1,7 +1,24 @@
 module Floe
   class Workflow
-    module IntrinsicFunction
-      def self.evaluate(payload, context = {}, input = {})
+    class IntrinsicFunction
+      def self.value(payload, context = {}, input = {})
+        new(payload).value(context, input)
+      end
+
+      def self.intrinsic_function?(payload)
+        Parser.new.parse(payload)
+        true
+      rescue Parslet::ParseFailed
+        false
+      end
+
+      attr_reader :payload
+
+      def initialize(payload)
+        @payload = payload
+      end
+
+      def value(context = {}, input = {})
```

Comment:
NOTE this is more in line how we handle Path types

Reply:
Yeah I noticed that when I started using Path, so this is much more consistent 👍